src/compiler/checker.ts

```diff
@@ -729,6 +728,10 @@ namespace ts {
         const keyofConstraintType = keyofStringsOnly ? stringType : stringNumberSymbolType;
         const numberOrBigIntType = getUnionType([numberType, bigintType]);
 
+        const identityMapper: TypeMapper = makeUnaryTypeMapper(anyType, anyType);

```

Comment:
`anyType` is an else case as a source and identity as a target?

Reply:
When the `source` and `target` are the same type, the mapper is basically a no-op. `anyType` is an arbitrary choice, any type would do.